ssz/sedes/serializable.py

```diff
@@ -0,0 +1,425 @@
+        """
+        Deserialize the data from the given start_index
+        """
+        # Make sure we have sufficient data for inferring length of container
+        if len(data) < start_index + CONTAINER_PREFIX_LENGTH:
+            raise DeserializationError(
+                'Insufficient data: Cannot retrieve the length of container',
+                data
+            )
+
+        # container_len contains the length of the original container
+        container_len = int.from_bytes(
+            data[start_index:start_index + CONTAINER_PREFIX_LENGTH],
+            'big'
+        )
+        container_end_index = start_index + CONTAINER_PREFIX_LENGTH + container_len
+
+        # Make sure we have sufficent data for inferring the whole container
+        if len(data) < container_end_index:
+            raise DeserializationError(
+                'Insufficient data: Cannot retrieve the whole container',
+                data
+            )
+
+        deserialized_field_values = []
+        field_start_index = start_index + CONTAINER_PREFIX_LENGTH
+        for _, field_sedes in cls._meta.fields:
+            field_value, next_field_start_index = field_sedes.deserialize_segment(
+                data,
+                field_start_index,
+            )
+            deserialized_field_values.append(field_value)
+            field_start_index = next_field_start_index
+
+        if field_start_index != container_end_index:
+            raise DeserializationError(
+                'Data to be deserialized is too long',
+                data
+            )
+
+        return tuple(deserialized_field_values), container_end_index
+
+    @classmethod
+    def deserialize(cls, data, **extra_kwargs):
+        # deserialized_field_values stores all the field values in tuple format
+        deserialized_field_values, end_index = cls.deserialize_segment(data, 0)
+        if end_index != len(data):
+            raise DeserializationError(
+                'Data to be deserialized is too long',
+                data
+            )
+
+        args_as_kwargs = merge_args_to_kwargs(deserialized_field_values, {}, cls._meta.field_names)
+        return cls(**args_as_kwargs, **extra_kwargs)
+
+    def copy(self, *args, **kwargs):
+        missing_overrides = set(
+            self._meta.field_names
+        ).difference(
+            kwargs.keys()
+        ).difference(
+            self._meta.field_names[:len(args)]
+        )
+        unchanged_kwargs = {
+            key: copy.deepcopy(value)
+            for key, value
+            in self.as_dict().items()
+            if key in missing_overrides
+        }
+        combined_kwargs = dict(**unchanged_kwargs, **kwargs)
+        all_kwargs = merge_args_to_kwargs(args, combined_kwargs, self._meta.field_names)
+        return type(self)(**all_kwargs)
+
+    def __copy__(self):
+        return self.copy()
+
+    def __deepcopy__(self, *args):
+        return self.copy()
+
+    _in_mutable_context = False
```

Comment:
I'm not seeing any code paths where this can be `True` ... am I missing something?

Reply:
You are right, I guess we aren't using this in a `mutable` context anywhere. @hwwhww could you please confirm this whether the `container datastructure` is needed to be `mutable` anywhere?

Reply:
@hwwhww I have pushed another commit assuming that it was to be `Immutable`. If this is not the case, I will revert back. Please let me know. Thankyou.